var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -180,10 +180,13 @@ def yes_no(varstr):
             options += ['MPICXX=%s' % spec['mpi'].mpicxx]
 
         if '+hypre' in spec:
-            options += [
-                'HYPRE_DIR=%s' % spec['hypre'].prefix,
-                'HYPRE_OPT=-I%s' % spec['hypre'].prefix.include,
-                'HYPRE_LIB=-L%s' % spec['hypre'].prefix.lib + ' -lHYPRE']
+            hypre = spec['hypre']
+            hypre_flag_list = [hypre.libs.ld_flags,
```

Comment:
`(hypre.libs+hypre['lapack'].libs+hypre['blas'].libs).ld_flags` IMO is better as it will deduplicated blas/lapack if needed.